src/devices/Rtc/SystemClock.cs

```diff
@@ -0,0 +1,264 @@
+        }
+
+        private static DateTime GetDateTimeUtcUnix()
+        {
+            string date = RunDateCommandUnix("-u +%s.%N", false, out _); // Floating point seconds since epoch
+            if (Double.TryParse(date, NumberStyles.Any, CultureInfo.InvariantCulture, out var result))
+            {
+                DateTime dt = UnixEpochStart + TimeSpan.FromSeconds(result);
+                return dt;
+            }
+
+            throw new IOException($"The return value of the date command could not be parsed: {date} (seconds since 01/01/1970)");
+        }
+
+        private static DateTime GetDateTimeUtcMacOs()
+        {
+            string date = RunDateCommandUnix("-u +%s", false, out _); // Floating point seconds since epoch
+            if (Double.TryParse(date, NumberStyles.Any, CultureInfo.InvariantCulture, out var result))
+            {
+                DateTime dt = UnixEpochStart + TimeSpan.FromSeconds(result);
+                return dt;
+            }
+
+            throw new IOException($"The return value of the date command could not be parsed: {date} (seconds since 01/01/1970)");
+        }
+
+        private static void SetDateTimeUtcUnix(DateTime dt)
+        {
+            string formattedTime = dt.ToString("yyyy-MM-dd HH:mm:ss.fffff", CultureInfo.InvariantCulture);
+            int exitCode;
+            // Try to run the date command as user first (maybe it has the set-user-id bit set) otherwise, try root.
+            RunDateCommandUnix($"-u -s \"{formattedTime}\"", false, out exitCode);
+            if (exitCode != 0)
+            {
+                var output = RunDateCommandUnix($"-u -s \"{formattedTime}\"", true, out exitCode);
+                if (exitCode != 0)
+                {
+                    throw new UnauthorizedAccessException($"Error running date command as local user and as root. Error {exitCode}: {output}");
+                }
+            }
+        }
+
+        private static void SetDateTimeUtcMacOs(DateTime dt)
+        {
+            // The format is "[[[[[cc]yy]mm]dd]hh]mm[.ss]" from https://ss64.com/osx/date.html - pretty weird to do this without delimiters
+            string formattedTime = dt.ToString("yyyyMMddHHmm.ss", CultureInfo.InvariantCulture);
+            int exitCode;
+            // Try to run the date command as user first (maybe it has the set-user-id bit set) otherwise, try root.
+            RunDateCommandUnix($"{formattedTime}", false, out exitCode);
+            if (exitCode != 0)
+            {
+                var output = RunDateCommandUnix($"{formattedTime}", true, out exitCode);
+                if (exitCode != 0)
+                {
+                    throw new UnauthorizedAccessException($"Error running date command as local user and as root. Error {exitCode}: {output}");
+                }
+            }
+        }
+
+        private static string RunDateCommandUnix(string commandLine, bool asRoot, out int exitCode)
```

Comment:
We should always run as current user and not try to elevate, if doesn't have permissions then it should fail, if possible use linux syscalls rather than starting new process.

Reply:
This might have some reference how to do it using syscalls https://stackoverflow.com/questions/28314543/set-system-date-and-time-using-c-in-linux

Reply:
Syscalls would be more complex than that, and they would require root priviledges. 

The above works fine on the Raspi or if the system is set up accordingly. I don't want that I need to run the program with root priviledges just to be able to set the time somewhere within. I've seen other GPS receiver applications do it similarly.